### ISSUE TITLE
Tracking issue for spark support

### ISSUE THREAD

**Issue by grief8** [asterinas/asterinas#1681]
This issue serves as a tracking log for running Spark on Asterinas. Before we begin, we need to configure the JVM. I have opted for JAVA8, using the following link: [JAVA8](https://cfdownload.adobe.com/pub/adobe/coldfusion/java/java8/java8u431/jdk/jdk-8u431-linux-x64.tar.gz). For Spark, I am utilizing the version available at: [Spark 3.5.3](https://archive.apache.org/dist/spark/spark-3.5.3/spark-3.5.3-bin-without-hadoop.tgz). Detailed usage instructions can be found in the [Asterinas JVM branch](https://github.com/grief8/asterinas/tree/jvm).

Currently, my tests are still ongoing. Occasionally, I can run the spark-example (i.e., spark-pi) successfully, but more often, the program terminates with SIGABRT, SIGBUS, or SIGSEGV signals. 

<details>

<summary>Runing results</summary>

![image](https://github.com/user-attachments/assets/89cf1a4a-f370-458f-b814-f9c476f17ed2)
![image](https://github.com/user-attachments/assets/b118943d-a60d-41a0-af7e-0b69f8fffc6b)

</details>

To summarize, based on my testing, Asterinas currently needs the following additions:

- **Necessary**:
  - [ ]  **bash (spark shell)**:
    - [ ]  `/usr/bin/env`
    - [ ]  `/usr/bin/bash`
    - [ ]  Shebang for one more argument
  - [x]  `sys_prlimit64` in #1654 
  - [ ]  `sys_setsocketopt(SOL_IP)`
  - [ ]  `sys_ioctl(SIOCGIFCONF | SIOCGIFFLAGS | SIOCGIFNETMASK | SIOCGIFINDEX | SIOCGIFBRDADDR)`
  - [ ] FPU registers
  
- **Not necessarily**:
  - [ ]  `sys_clone3(CLONE_SYSVSEM | pidfd)`
  - [ ]  `sys_madvise(MADV_DONTNEED)`
  - [ ]  `sys_rseq`
  - [ ]  `sys_sysinfo`
  - [ ]  `sys_clock_getres`

**Comment by tangruize** [comment]
`SIGBUS` and `SIGSEGV` may be related to #1652. There's a TODO for FPU save/restore in signals (see https://github.com/asterinas/asterinas/pull/1652#discussion_r1861408152). If Java's signal handler modifies FPU registers, it could cause `SIGBUS` and `SIGSEGV`. I'm not familiar with JVM or the Spark support, so there might be other causes.

**Comment by grief8** [comment]
> `SIGBUS` and `SIGSEGV` may be related to #1652. There's a TODO for FPU save/restore in signals (see [#1652 (comment)](https://github.com/asterinas/asterinas/pull/1652#discussion_r1861408152)). If Java's signal handler modifies FPU registers, it could cause `SIGBUS` and `SIGSEGV`. I'm not familiar with JVM or the Spark support, so there might be other causes.

I think it is likely to be this problem. A similar situation also occurs in #1608.

**Comment by grief8** [comment]
> > `SIGBUS` and `SIGSEGV` may be related to #1652. There's a TODO for FPU save/restore in signals (see [#1652 (comment)](https://github.com/asterinas/asterinas/pull/1652#discussion_r1861408152)). If Java's signal handler modifies FPU registers, it could cause `SIGBUS` and `SIGSEGV`. I'm not familiar with JVM or the Spark support, so there might be other causes.
> 
> I think it is likely to be this problem. A similar situation also occurs in #1608.

Hi @cqs21, could you give me some insights on this issue?

**Comment by cqs21** [comment]
> Hi @cqs21, could you give me some insights on this issue?

@grief8  I made a quick fix, but it still triggers `SIGSEGV`. I will take a closer look at this issue later.
